src/scalar.rs

```diff
@@ -1138,7 +1136,15 @@ impl Scalar {
     /// # }
     /// ```
     pub fn is_canonical(&self) -> Choice {
-        self.ct_eq(&self.reduce())
+        let mut over = Choice::from(0);
+        let mut under = Choice::from(0);
+        for (this, l) in self.unpack().0.iter().zip(&constants::L.0).rev() {
+            let gt = this.ct_gt(l);
+            let eq = this.ct_eq(l);
+            under |= (!gt & !eq) & !over;
+            over |= gt;
+        }
+        under
     }
```

Comment:
This seems like it should be fine.

Another way to do this check is to subtract the modulus and check for underflow, although I don't think that's going to be straightforward with `UnpackedScalar::sub`.

Reply:
That's an interesting idea!

I hacked your idea right now, and it seems to be the fastest yet :)

it drops to ~40ns 

I'll verify the code by hand and figure out what's the best place/naming for it but that's the logic I have in mind (and it passes the tests) https://godbolt.org/z/7jbqTxo6r

(sadly this would need to be added to the backends)